File: 프로그래머스/lv2/신은혜/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {    
    static boolean isPrime(long n){
        for(long i=2; i<=Math.sqrt(n); i++){
            if(n%i==0) return false;
        }
        return true;
    }
    
    public int solution(int n, int k) {
        String s = Integer.toString(n,k);
        String[] list = s.split("0");
        
        int cnt = 0;
        for(String num : list){
            if(num.equals("")) continue;
            long a = Long.parseLong(num);
            if(a!=1 && isPrime(a)){
                cnt++;
            }
        }
        
        return cnt;
    }
}

File: 프로그래머스/lv2/이하은/k진수에서 소수 개수 구하기.kt
/*
P는 0을 포함하지 않는 소수 
*/
class Solution {
    fun solution(n: Int, k: Int): Int {
        var answer = 0
        val nums = Integer.toString(n, k).split('0')
        
        for(e in nums){
            if(e == "") continue
            if(isPrime(e.toLong())) answer++
        }
        
        return answer
    }
    
    fun isPrime(x: Long): Boolean {
        if(x < 2) return false
        
        for(i in 2..x){
            if(i*i > x) break 
            if(x % i == 0L) return false 
        }
        
        return true 
    }
}